### PR TITLE
Check that a test named in tests/*.py's prose still exists

### LINKED ISSUE #258: Nothing checks that a test named in prose is still a test
#256 is about two landmarks in a module docstring that pointed at the
wrong test, and its fix names the tests instead of their position. That is
strictly better — an append cannot falsify a name — but it is not
*checked*: a rename falsifies it just as silently, and nothing here would
go red.

That is not hypothetical. #255's rename moved two test names, and the only
reason no prose went stale with them is that a human grepped for
citations; the changelog line that cites one of them was updated by hand in
that same diff. Test names in this suite are prose that other prose refers
to — the docstring of `tests/test_verified_signing.py` now names four of
them — so the class is growing rather than shrinking.

**What exists to check.** Every citation of the form `` `test_...` `` in
`tests/*.py`, which today is eleven of them across six modules:
`test_the_sweep_is_whole`, `test_pubkey_from_prvkey`,
`test_the_sender_wipes_the_keys_it_had_when_a_later_one_is_refused`,
`test_every_private_half_is_paired`,
`test_the_outputs_follow_the_recipient_order`, and the five in
`tests/test_verified_signing.py`. All eleven resolve to a test that exists
today — measured on d9f14de, so this issue is a guard against a defect
rather than a report of one.

**And the wrinkle that makes it a design question rather than three
lines.** One citation is not ours: `tests/test_vectors.py:36` and `:1154`
name `test_low_r`, which is rust-secp256k1's test in its own `src/lib.rs`,
cited to say which upstream vector is reproduced — correct prose that a
naive walk would report as a dangling reference. So the check has to tell a
citation of a local test from a citation of an upstream one, and the
options are not equal: matching only names that a test module defines
somewhere would silently pass a genuine typo, while excluding a list of
upstream names by hand is a list nothing maintains.

Where such a walk lives is the other half of the question. `tests/test_secret.py`
already walks the package for callers of `_secret.take` and is the
precedent for a test that reads the tree; a check over the *test* modules'
prose is a different population and may not belong beside it.

Noticed while fixing #256 and not addressed there: that diff is three
sentences of one docstring, and this is a check over every module's prose.

### PR DESCRIPTION
**Pull request by fametrano**
## Summary
- An append cannot falsify a `` `test_...` `` citation in `tests/*.py`'s docstrings and comments, but a rename does, silently -- #255's rename of two test names is why that is not hypothetical: only a human grepping for citations kept the prose in step with it at the time.
- Added `.github/scripts/check_test_citations.py`, which walks every docstring and comment in `tests/*.py` (never a module's own string literals -- see below) and flags a citation naming no test any module defines.
- Not every dangling-looking citation is a bug: `tests/test_vectors.py` correctly cites rust-secp256k1's own `test_low_r` to say which upstream vector is reproduced. A citation is read as another project's, and skipped, where the text right before it attributes it -- a possessive ("rust-secp256k1's `test_low_r`") or a foreign source file named in the same breath (`.c`/`.cpp`/`.rs`, this package's own tests being Python) -- rather than through a hand-maintained list of upstream test names, which is a list nothing forces anyone to update when a citation of a third project arrives.
- The walk is scoped to docstrings and `#` comments (via `ast` + `tokenize`), not every string literal a module contains, so a test module's own fixtures -- `tests/test_test_citations.py`'s own, built to look like citations without being read as one -- don't become a case for the thing they test.
- Wired in as an `always_run` local hook, for the same reason `submodule-pin` is one: the file a rename breaks a citation in is not necessarily the file the rename touches, so a `files` filter on what changed would miss exactly the case this exists for. Unlike `submodule-pin` it needs neither git nor the network, so it runs on pre-commit.ci unmodified.

## Test plan
- [x] `tests/test_test_citations.py` -- unit tests for `defined_tests`/`dangling_citations`/`main`, plus `test_the_real_suite_has_no_dangling_citation`, which runs the check against the real `tests/` tree (no monkeypatch needed) and is what proves the attribution heuristic covers `test_low_r` without a hand-maintained list of its name
- [x] `uv run --locked --no-default-groups --group test pytest --cov` -- 733 passed, 100% coverage
- [x] `uvx pre-commit run --all-files`, including the new hook against the real tree

Closes #258

## Summary by Sourcery

Add automated validation to keep test-suite prose citations synchronized with the tests they reference.

New Features:
- Add a check that validates `test_...` citations in test-suite docstrings and comments against defined tests.
- Recognize citations referring to tests from external projects without requiring a maintained exception list.

Bug Fixes:
- Detect stale test citations left behind by renamed or removed tests.

Enhancements:
- Limit citation analysis to Python docstrings and comments while resolving tests across the entire test suite.

CI:
- Run the test-citation validation as an always-run local pre-commit hook.

Tests:
- Add unit and real-suite coverage for citation resolution, external attribution, boundary cases, and command output.

### CODE REVIEW

File: .github/scripts/check_test_citations.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+# possessive naming whose test it is ("rust-secp256k1's `test_low_r`"),
+# or a source file foreign to this package's own tests -- which are
+# Python -- named in the same breath ("src/lib.rs\n`test_low_r`")
+_POSSESSIVE = re_compile(r"'s\s*$")
+_FOREIGN_SOURCE = re_compile(r"\.(?:c|cc|cpp|h|hpp|rs)\b")
+
+# far enough back to reach across one wrapped line -- a path ending one
+# line and the citation opening the next -- and no further: a citation
+# earns its own attribution, not one borrowed from a paragraph away
+_LOOKBACK = 200
```
[the remark this answers is not in the document]
Comment (fametrano) [reply .github/scripts/check_test_citations.py]: Fixed in 84763a4: both `_POSSESSIVE` and `_FOREIGN_SOURCE` are now anchored to sit right before the citation (only whitespace between), and `_POSSESSIVE` additionally requires a hyphen or digit in the possessive noun, so "the parser's" no longer reads as external attribution the way "rust-secp256k1's" does. Reproduced both gaps before fixing, and added `test_a_foreign_source_path_one_sentence_back_excuses_nothing` and `test_an_ordinary_possessive_is_not_read_as_external_attribution` as regression tests.

**Comment by fametrano** [comment]
Superseded by #261, which already closes #258 with a more complete implementation — it handles citations wrapped across line breaks (a case my regex here misses entirely) and also scans `btclib_secp256k1/*.py`'s own docstrings, not just `tests/*.py`. Closing this in favor of that one.
